e2etest/bgptests/node_selector.go

```diff
@@ -189,6 +189,81 @@ var _ = ginkgo.Describe("BGP Node Selector", func() {
 		table.Entry("IPV6 - one on first, two on second", ipfamily.IPv6, "fc00:f853:0ccd:e799::/116", []int{0}, []int{1, 2}),
 	)
 
+	table.DescribeTable("One service, one advertisement with node selector, labeling node",
+		func(pairingIPFamily ipfamily.Family, address string) {
+			var allNodes *corev1.NodeList
+			allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Setting advertisement with node selector (no matching nodes)")
+			resources := metallbconfig.ClusterResources{
+				Pools: []metallbv1beta1.IPAddressPool{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "test-pool",
+						},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: []string{address},
+						},
+					},
+				},
+				Peers: metallb.PeersForContainers(FRRContainers, pairingIPFamily),
+				BGPAdvs: []metallbv1beta1.BGPAdvertisement{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "test-adv",
+						},
+						Spec: metallbv1beta1.BGPAdvertisementSpec{
+							Communities:    []string{CommunityNoAdv},
+							IPAddressPools: []string{"test-pool"},
+							NodeSelectors: []metav1.LabelSelector{
+								{
+									MatchLabels: map[string]string{
+										"bgp-node-selector-test": "true",
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			for _, c := range FRRContainers {
+				err := frrcontainer.PairWithNodes(cs, c, pairingIPFamily)
+				framework.ExpectNoError(err)
+			}
+
+			err = ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			svc, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, "external-local-lb", testservice.TrafficPolicyCluster)
+			defer testservice.Delete(cs, svc)
+
+			ginkgo.By("Validating service IP not advertised")
+			checkServiceNotOnNodes(cs, svc, allNodes.Items, pairingIPFamily)
+
+			node := allNodes.Items[0]
```

Comment:
How about wrapping this labelling work in an util function under k8s ? Same for unlabeling I guess